src/process_wrap.cc

```diff
@@ -121,8 +123,10 @@ class ProcessWrap : public HandleWrap {
     // options.uid
     Local<Value> uid_v = js_options->Get(env->uid_string());
     if (uid_v->IsInt32()) {
-      int32_t uid = uid_v->Int32Value();
-      if (uid & ~((uv_uid_t) ~0)) {
+      int32_t uid_i32 = uid_v->Int32Value();
+      uv_uid_t uid = static_cast<uv_uid_t>(uid_i32);
+      // uv_uid_t may be unsigned, so compare with the widest available type
+      if (static_cast<intmax_t>(uid_i32) != static_cast<intmax_t>(uid)) {
```

Comment:
I don't think this logic is entirely correct (but neither is the existing logic.)  People sometimes pass e.g. `{ uid: -2 }` but this check would reject that.

(Also, there's a narrowing conversion two lines up when `sizeof(int32_t) > sizeof(uv_uid_t)`.  That's mostly academical, though.)